strapi/strapi
# Add path prefix option to s3 upload provider

### What does it do?

This PR adds an extra option to prefix the upload path for the s3 provider.

### Why is it needed?

This way people can upload to separate folders inside the same bucket based on their environment:
eg: 
development/file1.jpg
production/file1.jpg

### How to test it?

Make sure you have an environment variable set under: NODE_ENV
This env variable should be used when you add the corresponding configuration in the plugins.js file:
```
    upload: {
      provider: 'aws-s3',
      providerOptions: {
        pathPrefix: env('NODE_ENV'),
        accessKeyId: env('AWS_ACCESS_KEY_ID'),
        secretAccessKey: env('AWS_ACCESS_SECRET'),
        region: env('AWS_REGION'),
        params: {
          Bucket: env('AWS_BUCKET'),
        },
      },
    }
```
You can also test with a simple string:
```
    upload: {
      provider: 'aws-s3',
      providerOptions: {
        pathPrefix: 'my-folder',
        accessKeyId: env('AWS_ACCESS_KEY_ID'),
        secretAccessKey: env('AWS_ACCESS_SECRET'),
        region: env('AWS_REGION'),
        params: {
          Bucket: env('AWS_BUCKET'),
        },
      },
    }
```

### Related issue(s)/PR(s)

No, unrelated to any existing issues
